src/mods/storytimer.cpp

```diff
@@ -0,0 +1,368 @@
+        }
+        s_timer_group[k].split = sum[k] + s_timer_group[k].segment;
+    }
+    s_timer_group[0].split = s_timer_group[0].segment;
+    s_loadless_story_timer = s_timer_group[9].split;
+}
+
+void disp() {
+    if ((mkb::main_game_mode != mkb::STORY_MODE && mkb::sub_mode != mkb::SMD_AUTHOR_PLAY_INIT &&
+         mkb::sub_mode != mkb::SMD_AUTHOR_PLAY_MAIN) ||
+        freecam::should_hide_hud()) {
+        return;
+    }
+
+    bool is_between_worlds = false;
+    if ((s_completed_stages % 10 == 0) && s_completed_stages != 0 &&
+        mkb::sub_mode == mkb::SMD_GAME_GOAL_MAIN) {
+        is_between_worlds = true;
+    } else if ((s_completed_stages % 10 == 0 &&
+                mkb::g_storymode_stageselect_state == mkb::STAGE_SELECT_INTRO_SEQUENCE) ||
+               s_completed_stages % 10 != 0) {
+        // no longer "between worlds" if you enter the next worlds 10 ball screen, or if you break
+        // the tape on the last stage of the current world, but retry
+        is_between_worlds = false;
+    }
+
+    bool is_run_complete = (mkb::scen_info.world == 9 && s_completed_stages_world[9] == 10);
+
+    // move the positions of the fullgame if the death counter is on
+    u32 fullgame_timer_location_y = 2;
+    if (pref::get(pref::BoolPref::ShowDeathCounter)) {
+        fullgame_timer_location_y = 3;
+    } else {
+        fullgame_timer_location_y = 2;
+    }
+
+    bool display_story_timer = 0;
+    switch (TimerOptions(pref::get(pref::U8Pref::FullgameTimerOptions))) {
+        case TimerOptions::AlwaysShow:
+            display_story_timer = true;
+            break;
+        case TimerOptions::BetweenWorlds:
+            display_story_timer = is_between_worlds;
+            break;
+        case TimerOptions::EndOfRun:
+            display_story_timer = is_run_complete;
+            break;
+        case TimerOptions::DontShow:
+            display_story_timer = false;
+            break;
+    }
+
+    if (display_story_timer) {
+        timerdisp::draw_timer(FULLGAME_TIMER_LOCATION_X, fullgame_timer_location_y,
+                              FULLGAME_TIMER_TEXT_OFFSET, "Time:", s_loadless_story_timer, 0, false,
+                              false, draw::WHITE);
+    }
+
+    // move the position of the segment timer depending on if the death counter and fullgame timer
+    // is on; if the fullgame timer and death counter is off but the segment timer is on, move the
+    // segment timer to the top line; if either the fullgame timer or death counter are on but not
+    // both are on, move it to the 2nd line, if all 3 are enabled, put it on the 3rd line
+    u32 segment_timer_location_y = 2;
+    if (!display_story_timer && !pref::get(pref::BoolPref::ShowDeathCounter)) {
+        segment_timer_location_y = 2;
+    } else if (!display_story_timer || !pref::get(pref::BoolPref::ShowDeathCounter)) {
+        segment_timer_location_y = 3;
+    } else {
+        segment_timer_location_y = 4;
+    }
```

Comment:
A simpler way with less constant values:
```
u32 segment_timer_y = 2;
if (display_story_timer) { segment_timer_y++; }
if (pref::get(pref::BoolPref::ShowDeathCounter)) { segment_timer_y++; }
```

Reply:
The `timerdisp` refactor I was proposing would do this as well

Reply:
I just used rehtrop's suggestion, the timerdisp refactor will probably be the last thing I look into once everything else is resolved